therapy/etl/chembl.py

```diff
@@ -16,46 +16,34 @@
 class ChEMBL(Base):
     """ETL the ChEMBL source into therapy.db."""
 
-    def _extract_data(self, *args, **kwargs):
+    def _extract_data(self):
         """Extract data from the ChEMBL source."""
-        logger.info('Extracting chembl_27.db...')
-        if 'data_path' in kwargs:
-            chembl_db = kwargs['data_path']
+        self._src_data_dir.mkdir(exist_ok=True, parents=True)
+        chembl_files = list(self._src_data_dir.glob('chembl_*.db'))
+        if chembl_files:
+            chembl_path = sorted(chembl_files, reverse=True)[0]
+            search_results = re.search(r'chembl_([0-9]*).db', chembl_path.name)
+            self._version = search_results.group(1)
         else:
-            chembl_db = self._src_data_dir / 'chembl_27.db'
-        if not chembl_db.exists():
-            chembl_archive = self._src_data_dir / 'chembl_27_sqlite.tar.gz'
-            chembl_archive.parent.mkdir(exist_ok=True, parents=True)
             self._download_data()
-            tar = tarfile.open(chembl_archive)
-            tar.extractall(path=PROJECT_ROOT / 'data' / 'chembl')
-            tar.close()
-
-            # Remove unused directories and files
-            chembl_27_dir = self._src_data_dir / 'chembl_27'
-            temp_chembl = chembl_27_dir / 'chembl_27_sqlite' / 'chembl_27.db'
-            chembl_db = self._src_data_dir / 'chembl_27.db'
-            shutil.move(temp_chembl, chembl_db)
-            os.remove(chembl_archive)
-            shutil.rmtree(chembl_27_dir)
-        conn = sqlite3.connect(chembl_db)
+            chembl_path = list(self._src_data_dir.glob('chembl_*.db'))[0]
+        conn = sqlite3.connect(chembl_path)
         conn.row_factory = sqlite3.Row
         self._conn = conn
         self._cursor = conn.cursor()
-        assert chembl_db.exists()
-        logger.info('Finished extracting chembl_27.db.')
-
-    def _download_data(self, *args, **kwargs):
-        """Download ChEMBL data from FTP."""
-        logger.info(
-            'Downloading ChEMBL v27, this will take a few minutes.')
-        self._ftp_download('ftp.ebi.ac.uk',
-                           'pub/databases/chembl/ChEMBLdb/releases/chembl_27',
-                           self._src_data_dir,
-                           'chembl_27_sqlite.tar.gz')
 
-    def _transform_data(self, *args, **kwargs):
-        """Transform SQLite data to JSON."""
+    def _download_data(self):
+        """Download latest ChEMBL database file from EBI."""
+        os.environ['PYSTOW_HOME'] = str(self._src_data_dir.parent.absolute())
+        logger.info('Downloading latest ChEMBL version...')
+        tmp_path = chembl_downloader.download_extract_sqlite()
+        logger.info('ChEMBL download complete.')
+        self._version = chembl_downloader.latest()
+        shutil.move(tmp_path, self._src_data_dir)
+        shutil.rmtree(tmp_path.parent.parent.parent)
```

Comment:
Should `bioversions` and `bioregistry` directories also be removed? They're both empty

Reply:
Ooh maybe. I thought they might get used for caching the version numbers during runtime but not totally clear, I'll see what happens if you clear them out